robotPositioner/robotPositioner.h

```diff
@@ -0,0 +1,382 @@
+        m_vicon = new UDPClient<ObjectData>(m_vicon_udp_client_port);
+        m_viconCaptureControl = new CaptureControl( 
+            m_vicon_capture_control_ip_address,
+            m_capture_control_port,
+            m_vicon_capture_control_executable_path
+        );
+
+        // keep trying to get stream of data
+        while(m_vicon->getNumObjects() == 0) {
+            std::this_thread::sleep_for(std::chrono::seconds(1));
+            std::cout << "Waiting for object" << std::endl;
+        }
+
+        // start recording 
+        if(!m_viconCaptureControl->startRecording("test1")) {
+            std::cout << "failed to start recording" << std::endl;
+        } else {
+            std::cout << "vicon finished setting up" << std::endl;
+        }
+    }
+
+
+    // sets up the 360 SeeCam
+    void setupCamera360() {
+        //const std::string device = "/dev/video0";
+        const std::string device = m_video_device;
+
+        m_cam = new See3CAM_CU40(device, See3CAM_CU40::Resolution::_1280x720);
+        const cv::Size unwrapSize(180,50); // calculate camera input dimensions
+
+        const cv::Mat bubblescopeMask = See3CAM_CU40::createBubblescopeMask(m_cam->getSuperPixelSize());
+        m_cam->autoExposure(bubblescopeMask);
+        m_unwrapper = m_cam->createUnwrapper(unwrapSize);
+    }
+
+    // gets vicon data
+    double getViconData() {
+
+        auto objectData = m_vicon->getObjectData(0);
+   
+        BoBRobotics::Vector3<millimeter_t> translation = objectData.getPosition();
+        BoBRobotics::Vector3<radian_t> rotation = objectData.getAttitude();
+
+        m_pos_X = translation[0].value();
+        m_pos_Y = translation[1].value();
+        m_heading = rotation[0].value() * 180.0/PI;
+    }
+
+
+    // updates the range and bearing from the goal location
+    void updateRangeAndBearing() {
+
+        getViconData(); // update variables
+
+        double delta_x = m_goalPositionX - m_pos_X;
+        double delta_y = m_goalPositionY - m_pos_Y;
+        std::vector<double> delta;
+        delta.push_back(delta_x);
+        delta.push_back(delta_y);
+
+        // calculating the distance form the goal
+        double q = std::inner_product(std::begin(delta), std::end(delta), std::begin(delta), 0.0);
+        m_distanceFromGoal = sqrt(q);
+        m_bearingFromGoal = atan2(delta_y,delta_x)*180.0/PI - m_heading - 90.0;
+        
+        // changing from <0,360> to <-180, 180>
+        m_bearingFromGoal = angleWrapAround(m_bearingFromGoal);
+
+    } 
+
+    template <class T>
+    T angleWrapAround(T angle) {
+        // changing from <0,360> to <-180, 180>
+        if (angle <= 0) { angle += 360; } 
+        if (angle > 180) { angle -= 360; }
+        return angle;
+    }
+
+    template <class T, class S>
+    bool closeTo(const T num, const S threshold) {
+        if (num > -threshold && num < threshold) {
+            return true;
+        } else {
+            return false;
+        }
+    }
```

Comment:
This function and the one above should be ``static`` because they don't change the object.

For ``closeTo()`` you could just write:
```
return (num > -threshold && num < threshold);
```